packages/bee-q/src/components/toast/bq-toast.tsx

```diff
@@ -0,0 +1,135 @@
+import { h, Component, Element, Prop, Watch, Method, State, Host } from '@stencil/core';
+import { getColorCSSVariable, validatePropValue } from '../../shared/utils';
+import { TOAST_TYPE, TToastType } from './bq-toast.types';
+
+/**
+ * @part base - The component's internal wrapper of the Toast component.
+ * @part icon - `<div>` container element of toast icon component.
+ * @part text - `<div>` container element of toast text slot.
+ */
+@Component({
+  tag: 'bq-toast',
+  styleUrl: './scss/bq-toast.scss',
+  shadow: true,
+})
+export class BqToast {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+  @Element() el!: HTMLBqToastElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+  /** State of Toast */
+  @State() private shouldShowToast = false;
+
+  // Public Property API
+  // ========================
+  /** Type of Toast */
+  @Prop({ reflect: true, mutable: true }) type: TToastType = 'default';
+  /** Text color of Toast */
+  @Prop({ reflect: true }) textColor: string;
+  /** Should show icon of Toast */
+  @Prop({ reflect: true }) showIcon = false;
+  /** Should hide Toast after period of time */
+  @Prop({ reflect: true }) autoCloseTime: number;
+
+  // Prop lifecycle events
+  // =======================
+  @Watch('type')
```

Comment:
make sure the timer is always `>=0`
```typescript

      this.autoCloseTime = Math.max(0, this. autoCloseTime);

```

```suggestion
  @Watch('type')
  @Watch('autoCloseTime')
```